lib/src/provider/provider_data.dart

```diff
@@ -1,133 +1,97 @@
-// ignore_for_file: avoid_dynamic_calls, type_annotate_public_apis
-
-import 'package:flutter/cupertino.dart';
-import 'package:flutter/widgets.dart';
-import 'package:supabase_flutter/supabase_flutter.dart';
-import 'package:todo_app/api/api_key.dart';
+import 'package:flutter/material.dart';
 import 'package:todo_app/src/constants.dart';
+import 'package:todo_app/src/models/note.dart';
 
 ///
 class ProviderData with ChangeNotifier {
-  ///
-  // Note? notes;
-  bool loading = false;
-  final int _testd = 0;
+  bool _removeTaskBool = false;
+  Color _textColor = textColorK;
+///
+  List<Note> noteList = [];
+ ///
+  Note? noteInfo;
+ ///
+  List<Todo> taskListProvider = [];
 
   ///
-  int get getData => _testd;
-  // ignore: member-ordering-extended
-  List<Note> _listOfNotes = [];
-  // ignore: public_member_api_docs
-  List<Note> get getListOfNote => _listOfNotes;
+  bool get getRemoveTask => _removeTaskBool;
+  ///
+  Color get getTextColor => _textColor;
+   ///
+  List<Note> get getNoteList => noteList;
 
-  // ignore: member-ordering-extended
-  List<Todo> _listOfTask = [];
+   ///
+  List<Todo> get getTaskList => taskListProvider;
+  
 
   ///
-  List<Todo> get getListOfTask => _listOfTask;
-// ignore: member-ordering-extended
-  ///
-  final client = SupabaseClient(supabaseUrl, supabaseKey);
+  Note? get getNoteInfo => noteInfo;
 
   ///
-  ProviderData();
+  void remove() {
+    _removeTaskBool = !_removeTaskBool;
+    notifyListeners();
+  }
```

Comment:
```suggestion
  void toggleRemoveTask() {
    _removeTaskBool = !_removeTaskBool;
    notifyListeners();
  }
```

Reply:
Besides, it looks like this logic can be moved to a Widget instead.

A Provider should only receive notification of the final picked option -- to remove or not.